### PR TITLE
feat: Add schema comparision to the main request loop for multiplexing to correctly update schema 

### PR DESCRIPTION
**Pull request by GaoleMeng**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigquerystorage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java [hunk 1]
```diff
@@ -457,7 +457,6 @@ private void appendLoop() {
             && !streamName.isEmpty()
             && !originalRequest.getWriteStream().equals(streamName)) {
           streamName = originalRequest.getWriteStream();
-          writerSchema = originalRequest.getProtoRows().getWriterSchema();
```
Comment (yirutang) [inline google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java:460]: Wondering why is this named originalRequest? Should it be currentRequest?

File: google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/StreamWriterTest.java [hunk 2]
```diff
@@ -562,6 +566,89 @@ public void testOneMaxInflightRequests_MultiplexingCase() throws Exception {
...
+    List<ApiFuture<AppendRowsResponse>> futures = new ArrayList<>();
+    // In total insert append `appendCountPerStream` * 2 requests.
+    for (int i = 0; i < appendCountPerStream; i++) {
+      futures.add(writer1.append(createProtoRows(new String[] {String.valueOf(i)}), i * 2));
+      futures.add(writer2.append(createProtoRows(new String[] {String.valueOf(i)}), i * 2 + 1));
+    }
+    for (int i = 0; i < appendCountPerStream * 2; i++) {
+      AppendRowsRequest appendRowsRequest = testBigQueryWrite.getAppendRequests().get(i);
+      assertEquals(i, appendRowsRequest.getOffset().getValue());
+      if (i == 0) {
```
Comment (yirutang) [inline google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/StreamWriterTest.java:639]: check stream name?

**Review by yirutang — COMMENTED** [review]
Looks good, some nits.

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java (hunk 1 above)
Comment (GaoleMeng) [reply google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java:460]: The originalRequest represent the request content before modifying,
the currentRequest lacks this level of meaning. So I feel like originalRequest is more close to the meaning here.

File: google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/StreamWriterTest.java (hunk 2 above)
Comment (GaoleMeng) [reply google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/StreamWriterTest.java]: Done,

**Review by yirutang — APPROVED** [review]

**Pull request merged**